src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -381,159 +382,16 @@ public BulkLoadRegisterSDFResponseDTO registerSdf(BulkLoadRegisterSDFRequestDTO
-					logError(e, numRecordsRead, mol, mappings, errorMolExporter, results, errorCSVOutStream);
-					continue;
-				}
-				lot.setSaltForm(saltForm);
-				lot.setParent(parent);
-
-				if (lot.getId() == null)
-					lot.setBulkLoadFile(bulkLoadFile);
-				if (saltForm.getId() == null)
-					saltForm.setBulkLoadFile(bulkLoadFile);
-				if (parent.getId() == null)
-					parent.setBulkLoadFile(bulkLoadFile);
-
-				try {
-					lot = validateLot(lot, mappings, results);
-				} catch (Exception e) {
-					logError(e, numRecordsRead, mol, mappings, errorMolExporter, results, errorCSVOutStream);
-					continue;
-				}
-
-				Metalot metalot = new Metalot();
-				metalot.setLot(lot);
-				metalot.setIsosalts(lot.getSaltForm().getIsoSalts());
-				metalot.setSkipParentDupeCheck(true);
-				MetalotReturn metalotReturn = null;
-
-				if (validate) {
-					try {
-						parent = validateParentAgainstDryRunCompound(parent, numRecordsRead, results);
-						// Check list of aliases within file being bulkloaded
-						if (!propertiesUtilService.getAllowDuplicateParentAliases()) {
-							for (ParentAlias alias : parent.getParentAliases()) {
-								if (allAliasMaps.get(alias.getAliasName()) == null) {
-									allAliasMaps.put(alias.getAliasName(), numRecordsRead);
-								} else {
-									throw new NonUniqueAliasException(
-											"Within File, Parent Alias " + alias.getAliasName()
-													+ " is not unique");
-								}
-							}
-						}
-						saveDryRunCompound(mol, parent, numRecordsRead, dryRunCompound);
-
-					} catch (TransactionSystemException rollbackException) {
-						logger.error("Rollback exception", rollbackException.getApplicationException());
-						Exception causeException = new Exception(
-								rollbackException.getApplicationException().getMessage(),
-								rollbackException.getApplicationException());
-						logError(causeException, numRecordsRead, mol, mappings, errorMolExporter, results,
-								errorCSVOutStream);
-						continue;
-					} catch (Exception e) {
-						logError(e, numRecordsRead, mol, mappings, errorMolExporter, results, errorCSVOutStream);
-						continue;
-					}
-
-				}
-				if (!validate) {
-					try {
-						metalotReturn = metalotService.save(metalot);
-						if (!metalotReturn.getErrors().isEmpty()) {
-							for (ErrorMessage errorMessage : metalotReturn.getErrors()) {
-								Exception e = new Exception(errorMessage.getMessage());
-								logError(e, numRecordsRead, mol, mappings, errorMolExporter, results,
-										errorCSVOutStream);
-							}
-							continue;
-						}
-					} catch (Exception e) {
-						logError(e, numRecordsRead, mol, mappings, errorMolExporter, results, errorCSVOutStream);
-						continue;
+				try{
+					Boolean isNewParent = registerMol(mol, bulkLoadFile, mappings, errorCSVOutStream, registeredCSVOutStream, errorMolExporter, registeredMolExporter, allAliasMaps, numRecordsRead, results, validate, chemist, registerRequestDTO, dryRunCompound, startTime);
```

Comment:
This was the major refactor to fix our transaction issue.  registerMol is wrapped in a new transaction the "next loop" can catch the saved structure from the first loop. Additionally, if an error is hit during registerMol, it doesn't roll back the whole transactions (which was the case before the un-released 1.13.7.8